library/src/webapp/js/portal/portal.theme.switcher.js

```diff
@@ -0,0 +1,71 @@
+var portal = portal || {};
+
+portal.darkThemeSwitcher = document.getElementById("sakai-dark-theme-switcher");
+
+portal.defaultThemeClass = 'sakaiUserTheme-notSet';
+portal.lightThemeClass = 'sakaiUserTheme-light';
+portal.darkThemeClass = 'sakaiUserTheme-dark';
+
+portal.toggleDarkTheme = () => {
+  // toggle the dark theme switch to the opposite state
+  portal.darkThemeSwitcher && portal.darkThemeSwitcher.getAttribute("aria-checked") === "false" ? portal.enableDarkTheme() : portal.enableLightTheme();
+};
+
+portal.setDarkThemeSwitcherToggle = onOff => {
+  portal.darkThemeSwitcher && portal.darkThemeSwitcher.setAttribute("aria-checked", onOff);
+};
+
+portal.addCssClassToMarkup = themeClass => document.documentElement.classList.add(themeClass);
+
+portal.removeCssClassFromMarkup = themeClass => document.documentElement.classList.remove(themeClass);
+
+portal.isOsDarkThemeSet = () => window.matchMedia('(prefers-color-scheme: dark)').matches;
+
+portal.setPortalThemeUserPref = theme => {
+
+  const url = `/direct/userPrefs/updateKey/${portal.user.id}/sakai:portal:theme?theme=${theme}`;
+  const ajaxRequest = new XMLHttpRequest();
+  ajaxRequest.open("PUT", url);
+  ajaxRequest.send();
+};
```

Comment:
```suggestion
portal.setPortalThemeUserPref = async theme => {

  const url = `/direct/userPrefs/updateKey/${portal.user.id}/sakai:portal:theme?theme=${theme}`;
  
  try {
    const response = await fetch(url, { method: 'PUT' });
    
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
  } catch (error) {
    console.error('There was a problem with the request:', error);
  }
};
```

Reply:
I don't like async, tbh. It tries to make asynchronous code look like synchronous code. promise.then is a lot clearer in intent, I think.